punishments.js

```diff
@@ -587,9 +591,9 @@ Punishments.roomPunish = function (room, user, punishment, recursionKeys) {
 			keys: Array.from(keys),
 			punishType: punishType,
 			rest: rest,
-		}, room.id + ':' + id, ROOM_PUNISHMENT_FILE);
+		}, roomid + ':' + id, ROOM_PUNISHMENT_FILE);
 
-		if (!(room.isPrivate === true || room.isPersonal || room.battle)) Punishments.monitorRoomPunishments(user);
+		if (roomid === "battle" || !(room.isPrivate === true || room.isPersonal || room.battle)) Punishments.monitorRoomPunishments(user);
```

Comment:
TypeScript isn't complaining about this? This should probably be `typeof room === 'string' ||`